src/ledger/grain.js

```diff
@@ -185,4 +194,87 @@ export function toFloatRatio(numerator: Grain, denominator: Grain): number {
   return Number(numerator) / Number(denominator);
 }
 
+/**
+ * Splits a budget of Grain into fractional shares.
+ *
+ * It guarantees that the sum of all the splits will be exactly equal
+ * to the budget. It does not guarantee fairness, i.e. some users may
+ * receive more or less Grain than others with equal shares.
+ */
+export function splitBudget(
+  budget: Grain,
+  shares: $ReadOnlyArray<number>
```

Comment:
Shares = cred balance? Have we used this wording anywhere else before? Why not call it `credBalances` or something like that?

Reply:
This is a more abstract function for partitioning a bigInt into pieces proportional to floating point shares. The shares aren't necessarily Cred--if you look at the usage in the grain allocation module, in one case the shares represent Cred, in another they represent per-user fractions of the total underpayment.

Reply:
I renamed it to `scores` which will hopefully be clearer.